cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DistributedSchedulerService.java

```diff
@@ -78,6 +82,7 @@ protected void doStop() {
   protected void startUp() throws Exception {
     LOG.info("Starting scheduler.");
     serviceDelegate.startAndWait();
+    countDownLatch.await();
```

Comment:
@poornachandra So this is the service that App-Fabric would call?? Meaning this is actually blocking app-fabric?? Have you tried while starting master, but before this block returns, issue a stop to master?? Because effectively that would kill the yarn app, making the dataset service never coming up. Will this latch block forever then??